src/core/main.c

```diff
@@ -952,12 +966,9 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case ARG_CONFIRM_SPAWN:
-                        r = optarg ? parse_boolean(optarg) : 1;
-                        if (r < 0) {
-                                log_error("Failed to parse confirm spawn boolean %s.", optarg);
-                                return r;
-                        }
-                        arg_confirm_spawn = r;
+                        r = parse_confirm_spawn(optarg ? : "y", &arg_confirm_spawn);
```

Comment:
i think it would be nicer to make parse_confirm_spawn() deal with the actual argument of NULL here, instead of "working around" it crashing on NULL by synthesizing fake input of "y" here...